torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -303,15 +304,234 @@ void LoopNestGenerator::handle(Expr* expr) {
     popFor();
 }
 
+namespace {
+
+TensorView* findOutputTensor(Expr* expr) {
+  if (expr->outputs().size() != 1) {
+    return nullptr;
+  }
+  auto out = expr->output(0);
+  if (out->getValType() != ValType::TensorView) {
+    return nullptr;
+  }
+  return out->as<TensorView>();
+}
+
+void findTargetTensor(Expr* expr, TensorView*& target, unsigned& score) {
+  TORCH_INTERNAL_ASSERT(expr->outputs().size() <= 1);
+
+  TensorView* out_tv = findOutputTensor(expr);
+  if (out_tv == nullptr) {
+    target = nullptr;
+    score = 0;
+    return;
+  }
+
+  if (!out_tv->hasComputeAt()) {
+    target = out_tv;
+    // No computeAt, so this should come last.
+    score = std::numeric_limits<unsigned>::max();
+    return;
+  }
+
+  auto axis = out_tv->getRelativeComputeAtAxis();
+  target = out_tv->getComputeAtView();
+  while (target->hasComputeAt()) {
+    if (axis <= target->getThisComputeAtAxis()) {
+      // Merge this expr to the target of the target group
+      axis = target->getRelativeComputeAtAxis() -
+          (target->getThisComputeAtAxis() - axis);
```

Comment:
I think there are some assumptions here. What I think you're trying to do is propagate this axis forward through targets to see where it ends up. However, I think you could just use: The TensorView function:
```.cpp
  // Will check if an axis is inside computeAtAxis and will fetch the reference
  // to be used in code generation.
  std::pair<IterDomain*, TensorView*> getComputeAtAxis(int pos) {
    TORCH_INTERNAL_ASSERT(
        nDims() > 0, "Tried to access a computeAt axis in a 0-dim TensorView");
    if (!hasComputeAt() || getThisComputeAtAxis() <= (unsigned int)pos)
      return std::pair<IterDomain*, TensorView*>(axis(pos), this);
    return compute_at_view_->getComputeAtAxis(getComputeAtRelPos(pos));
  }
```

Reply:
Thanks for the comment. That seems basically the same what the code does, though I need a small change of `getComputeAxis`. For my case, I need the index of the axis rather than `axis(pos)`. Would you mind changing `getComputeAxis` to return just `pos` instead of `axis(pos)`?

Reply:
That would be fine, would you like to do it in this PR or would you like me to do it in a follow up?

Reply:
Done. I ended up keeping the signature of `getComputeAtAxis` and adding another function that returns `<int, TensorView*>` because there are many places using just `IterDomain`, so having a function returning that makes sense.